data/src/main/java/io/homeassistant/companion/android/data/integration/IntegrationRepositoryImpl.kt

```diff
@@ -143,18 +185,23 @@ class IntegrationRepositoryImpl @Inject constructor(
         return retVal.toTypedArray()
     }
 
-    private fun createRegisterDeviceRequest(deviceRegistration: DeviceRegistration): RegisterDeviceRequest {
+    private suspend fun createRegisterDeviceRequest(deviceRegistration: DeviceRegistration): RegisterDeviceRequest {
+        val oldDeviceRegistration = getRegistration()
         return RegisterDeviceRequest(
             deviceRegistration.appId,
             deviceRegistration.appName,
-            deviceRegistration.appVersion,
-            deviceRegistration.deviceName,
-            deviceRegistration.manufacturer,
-            deviceRegistration.model,
+            deviceRegistration.appVersion ?: oldDeviceRegistration.appVersion,
+            deviceRegistration.deviceName ?: oldDeviceRegistration.deviceName,
+            deviceRegistration.manufacturer ?: oldDeviceRegistration.manufacturer,
```

Comment:
This seems weird. Why not just refer to the original source of this data in this function ?

Reply:
The request requires all data be passed, however it seemed strange to have to send all the data every time down the whole stack.  It seemed simpler to persist the values and only send potential deltas on update rather than the whole request.  For example if I'm changing device name, why should I need to generate the appData with the push url and token.

Reply:
What I meant was, why do we need all data to be passed ? Isn't this value hardcoded per device ?

Reply:
Yes, but the way we have our application structured we don't have access to that in the libraries.  We might be able to inject them via Dagger?  I don't claim to be an expert on that, but I can look into it.

Reply:
Injected now.